#! /bin/sh -e

# DP: Enable db compatibility add-on to be built

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

diff -urN glibc-2.1.94.orig/Makeconfig glibc-2.1.94/Makeconfig
--- glibc-2.1.94.orig/Makeconfig	Mon Sep 25 11:23:01 2000
+++ glibc-2.1.94/Makeconfig	Fri Sep 29 14:01:25 2000
@@ -822,7 +823,7 @@
 # is more or less arbitrary.  The sorting step will take care of the
 # dependencies.  Only the $(binfmt-subdir) should always be kept at the
 # end of the list.
-all-subdirs = csu assert ctype locale intl catgets math setjmp signal	    \
+all-subdirs = csu assert ctype locale intl catgets math setjmp signal db   \
 	      stdlib stdio-common $(stdio) malloc string wcsmbs time dirent \
 	      grp pwd posix io termios resource misc socket sysvipc gmon    \
 	      gnulib iconv iconvdata wctype manual shadow po argp	    \
diff -urN glibc-2.1.94.orig/db/Makefile glibc-2.1.94/db/Makefile
--- glibc-2.1.94.orig/db/Makefile	Sun Mar 21 20:14:47 1999
+++ glibc-2.1.94/db/Makefile	Fri Sep 29 14:18:06 2000
@@ -41,6 +39,8 @@
 
 $(db1-headers:%=$(inst_includedir)/db1/%): $(inst_includedir)/db1/%: % $(+force)
 	$(do-install)
+
+libdb1-inhibit-o = $(filter-out .os,$(object-suffixes))
 
 include ../Rules
 
diff -urN glibc-2.1.94.orig/shlib-versions glibc-2.1.94/shlib-versions
--- glibc-2.1.94.orig/shlib-versions	Mon Sep 25 11:23:06 2000
+++ glibc-2.1.94/shlib-versions	Fri Sep 29 14:01:25 2000
@@ -98,6 +98,11 @@
 mips.*-.*-linux.*	libnsl=1		GLIBC_2.0 GLIBC_2.2
 .*-.*-.*		libnsl=1
 
+# This is the Berkeley DB 1.85.  We use the version numbers from glibc 2.0.*
+# for all times.
+alpha-.*-linux.*        libdb1=2.1
+.*-.*-.*                libdb1=2
+
 # This defines the shared library version numbers we will install.
 alpha.*-.*-linux.*	libcrypt=1.1
 .*-.*-.*		libcrypt=1
--- glibc-2.1.94.orig/db/ndbm.h	Mon Jan 29 11:03:28 2001
+++ glibc-2.1.94/db/ndbm.h	Mon Jan 29 11:03:06 2001
@@ -39,7 +39,7 @@
 #ifndef _NDBM_H
 #define	_NDBM_H 1
 
-#include <db.h>
+#include <db1/db.h>
 
 /* Map dbm interface onto db(3). */
 #define DBM_RDONLY	O_RDONLY
